src/elements/input/input.svelte

```diff
@@ -95,12 +95,23 @@ const handleInput = () => {
     return;
   }
 
-  if (type === 'number' && input.value.endsWith('.')) {
-    return;
+  if (type === 'number') {
+    // ensure input of chars that form valid numbers
+    let numString = '';
+    for (const c of input.value) {
+      if (/[\d+.e-]/.test(c)) {
+        numString += c;
+      }
+    }
+    input.value = value = numString;
+
+    // only send value if formatted as valid number
+    if (Number.isNaN(Number.parseFloat(value))) {
```

Comment:
have you checked this doesn't throw an exception on weird input types?

Reply:
the weirdest thing about this function is that it'll greedily parse the float for a valid number. so on input `"1..."` returns as valid, and it'll parse `1`. similarly, `1...1` will parse as `1`. `1.1.1` parses as `1.1`

[`Number.parseFloat()` returns `NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseFloat) if the string passed in is not a number at all (ex. "...." or "+" etc)

we should probably expand this validation for `type=integer`

Reply:
oh actually, looks like we can use `Number()` if we want this check to be stricter
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat#description

Reply:
decided to use `Number()` to align user experience